streamlit_web.py

```diff
@@ -652,5 +657,38 @@ def interactive():
         st.write(summaries)
 
 
+def entities():
+    """Page to display entity analysis"""
+    st.header("Entity analysis inspects the given text for known entities \
+    and returns information about those entities. It is a way to extract \
+    information that seeks to locate and classify named entities in text \
+    into pre-defined categories such as the names of persons, organizations, \
+    locations, expressions of times, quantities, monetary values, and percentages.")
+    st.sidebar.success("Hello sidebar!")
+
+    # make a copy of the main dataframe
+    input_df = main_df.copy(deep=True)
+
+    # list out possible user documents to choose from and select one
+    students = st.multiselect(
+        label="Select specific students below:",
+        options=input_df[stu_id].unique(),
+    )
+
+    # create a dataframe with the selected user and convert it to a string
+    df_selected_stu = input_df.loc[input_df[stu_id].isin(students)]
+    student_string = df_selected_stu.to_string()
+
+    # run the spacy entity recogonizer on the selected user document and display it
+    if len(students) != 0:
+        docff = az.get_nlp(student_string)
+        html = spacy.displacy.render(docff, style="ent")
+        html = html.replace("\n", " ")
+        HTML_WRAPPER = """<div style="overflow-x: auto; border: 1px solid \
+    #e6e9ef; border-radius: 0.25rem; padding: 1rem; margin-bottom: 2.5rem">\
+    {}</div>"""
+        st.write(HTML_WRAPPER.format(html), unsafe_allow_html=True)
```

Comment:
This segment is largely the repetitive code from line 640 to 651. Could your team make it into a function so it's reuseable and more maiatainable?